pkg/orchestration/wiring/datadog/types.go

```diff
@@ -0,0 +1,57 @@
+package datadog
+
+import (
+	"github.com/atlassian/voyager"
+)
+
+type AlarmSpecType string
+type AlarmType string
+
+const (
+	Metric AlarmSpecType = "metric alert"
+)
+const (
+	CPU    AlarmType = "cpu"
+	Memory AlarmType = "memory"
+)
+
+type Alarm struct {
+	Name    string      `json:"name"`
+	Type    string      `json:"type"`
+	Query   string      `json:"query"`
+	Option  AlarmOption `json:"options"`
+	Message string      `json:"message"`
+}
+
+type ServiceInstanceSpec struct {
+	ServiceName voyager.ServiceName `json:"serviceName"`
+	Attributes  Alarm               `json:"attributes"`
+	Environment voyager.EnvType     `json:"environment"`
+	Region      voyager.Region      `json:"region"`
+}
+type AlarmOption struct {
+	Timeout           int             `json:"timeout_h,omitempty"`
```

Comment:
As far as I understand, each of the fields there are optional. To make it more explicit, I would consider making all of them pointers, i.e.
```
type AlarmOption struct {
  Timeout *int
  Locked *bool
  ...
}
```
But that may lead to more boilerplate :(